scripts/docker_build/sof_builder/Dockerfile

```diff
@@ -89,8 +91,10 @@ ENV PATH="/home/sof/work/xtensa-cnl-elf/bin:${PATH}"
 ENV PATH="/home/sof/work/xtensa-imx-elf/bin:${PATH}"
 ENV PATH="/home/sof/work/xtensa-imx8m-elf/bin:${PATH}"
 
-RUN cd /home/sof && git clone https://github.com/jcmvbkbc/newlib-xtensa.git newlib-xtensa.git && \
-	cd newlib-xtensa.git && git checkout -b xtensa origin/xtensa && \
+ARG NEWLIB_REPO=https://github.com/jcmvbkbc/newlib-xtensa.git
```

Comment:
Is it possible to group all these near the top to ease local mirroring? Disclaimer: this is the first Dockerfile I look at.

Reply:
The ARG command, just like the others, creates a new layer. This one just changing the environment the main process of the container is starting with. Probably we could move it near (but not at) the top, just we shouldn't have anything before the FROM statement.

Reply:
ARG command should be near where it is used to avoid rebuilding of all the layers that were created prior to that.

Quoting this answer : https://stackoverflow.com/a/41593407/5830996
> If you change the value of a build argument all of the layers after that ARG line will be invalidated. So I guess you should include it just before you use the ARG. 


And ARG by itself doesn't create any layers in the final image. Also answers your comment regarding introducing variables..

Quoting from docs: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers
> Only the instructions RUN, COPY, ADD create layers. Other instructions create temporary intermediate images, and do not increase the size of the build.